TwMapView.js

```diff
@@ -11,30 +12,128 @@ const styles = StyleSheet.create({
   }
 });
 
+/**
+* @class TwMapView
+* display of data on a map
+*/
 export class TwMapView extends React.Component {
   state = {
     region: {
       latitude: 53.350140,
       longitude: -6.266155,
       latitudeDelta: 0.2,
       longitudeDelta: 0.2
-    }
+    },
+    markers: []
   }
 
+  /**
+  * @function componentDidMount
+  * function that is called after the component is initialised
+  * this calculates the initial list of markers
+  */
+  componentDidMount() {
+    this._findLocalMarkers();
+  }
+
+  /**
+  * @function onRegionChange
+  * this gets called when the user moves the map
+  */
   onRegionChange(region) {
-    this.setState({ region });
+    this.setState({ region: region });
+    this._findLocalMarkers();
+  }
+
+  /**
+  * @function _findLocalMarkers
+  * calculates the maximum/minimum points for the screen display
+  * only adds notices to the list if they are within these boundaries
+  * this increases loading speed
+  * whenever the setState is called and the markers item is updated,
+  * the markers shown on the map are updated automatically
+  */
+  _findLocalMarkers() {
+    const regionInfo = this.state.region;
+    const allNotices = this.props.data;
+    const localNotices = [];
+    const latDelta = regionInfo.latitudeDelta;
+    const lngDelta = regionInfo.longitudeDelta;
+    const maxLatitude = regionInfo.latitude + latDelta;
+    const minLatitude = regionInfo.latitude - latDelta;
+    const maxLongitude = regionInfo.longitude + lngDelta;
+    const minLongitude = regionInfo.longitude - lngDelta;
+
+    allNotices.forEach((notice) => {
+      const lat = notice.LAT;
+      const lng = notice.LONG;
+
+      if (lat > minLatitude &&
+        lat < maxLatitude &&
+        lng > minLongitude &&
+        lng < maxLongitude) {
+        localNotices.push(notice);
+      }
+    });
+
+    console.log('No of Markers: ', localNotices.length);
```

Comment:
We can come back later and tidy up all these console logs

Reply:
I kept them in there for now so I can see how often this method is called and to know how many poles it has found.